vio/viosslfactories.c

```diff
@@ -238,8 +386,26 @@ new_VioSSLFd(const char *key_file, const char *cert_file,
     NOTE: SSL_CTX_set_cipher_list will return 0 if
     none of the provided ciphers could be selected
   */
-  if (cipher &&
-      SSL_CTX_set_cipher_list(ssl_fd->ssl_context, cipher) == 0)
+  strcat(cipher_list, tls_cipher_blocked);
+  if (cipher)
+  {
+    if (strlen(cipher_list) + strlen(cipher) + 1 > sizeof(cipher_list))
+    {
+      *error= SSL_INITERR_CIPHERS;
```

Comment:
Definitely need
`SSL_CTX_free(ssl_fd->ssl_context);` before `my_free(ssl_fd);`

Probably, add
`DBUG_PRINT("error", ("SSL cipher is too long: %s", sslGetErrString(*error)));`
`report_errors();`
as well